apps/crn-server/src/handlers/external-author/index-handler.ts

```diff
@@ -70,18 +87,48 @@ const squidexGraphqlClient = new SquidexGraphql(getAuthToken, {
   baseUrl,
 });
 
+const contentfulGraphQLClient = getContentfulGraphQLClient({
+  space: contentfulSpaceId,
+  accessToken: contentfulAccessToken,
+  environment: contentfulEnvId,
+});
+
 /* istanbul ignore next */
-export const handler = sentryWrapper(
-  indexExternalAuthorHandler(
-    new ExternalAuthorSquidexDataProvider(
+const getContentfulRestClientFactory = () =>
+  getContentfulRestClient({
+    space: contentfulSpaceId,
+    accessToken: contentfulManagementAccessToken,
+    environment: contentfulEnvId,
+  });
+
+const externalAuthorDataProvider = isContentfulEnabledV2
+  ? new ExternalAuthorContentfulDataProvider(
+      contentfulGraphQLClient,
+      getContentfulRestClientFactory,
+    )
+  : new ExternalAuthorSquidexDataProvider(
       externalAuthorRestClient,
       squidexGraphqlClient,
-    ),
-    algoliaSearchClientFactory({ algoliaApiKey, algoliaAppId, algoliaIndex }),
+    );
+
+/* istanbul ignore next */
+const selectedAlgoliaIndex = isContentfulEnabledV2
+  ? `${algoliaIndex}-contentful`
+  : algoliaIndex;
```

Comment:
actually i don't think this is valid, this is sort of why this ticket is blocked by mine IF we want to add the live resource syncing with algolia for contentful. 

As it is proposed in this PR regardless of where the event was emitted from it would be indexed based on the feature flag - thats actually incorrect, right? We want to index in the  `-contentful` index if it came from Contentful.